File: Forge/src/main/java/hantonik/atomic/core/utils/AtomicFormatting.java
package hantonik.atomic.core.utils;

import java.text.DecimalFormat;

public final class AtomicFormatting {
    public static String formatTime(long time) {
        var formattedTime = new StringBuilder();

        var ticks = time % 20;
        var seconds = (time / 20) % 60;
        var minutes = ((time / 20) % 3600) / 60;
        var hours = (time / 20) / 3600;

        if (hours > 0)
            formattedTime.append(hours).append("h ");

        if (minutes > 0)
            formattedTime.append(minutes).append("m ");

        if (seconds > 0 && hours <= 0)
            formattedTime.append(seconds).append("s ");

        if (seconds <= 0 && minutes <= 0 && hours <= 0)
            formattedTime.append(ticks).append("t ");

        return formattedTime.toString().trim();
    }

    public static String format(long input) {
        var pattern = new DecimalFormat("#.##");
        String[] suffixes = {" ", " k", " M", " G", " T"};

        for (String suffix : suffixes) {
            if (input >= 1000)
                input /= 1000;
            else
                return pattern.format(input) + suffix;
        }

        return pattern.format(input) + suffixes[suffixes.length - 1];
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/util/AtomicFormatting.kt
package hantonik.atomic.core.util

import java.text.DecimalFormat

object AtomicFormatting {
    fun formatTime(time: Long): String {
        val formattedTime = StringBuilder()

        val ticks = time % 20
        val seconds = (time / 20) % 60
        val minutes = (time / 20 / 60) % 60
        val hours = (time / 20 / 60 / 60) % 60

        if (hours > 0)
            formattedTime.append(hours).append("h ")

        if (minutes > 0)
            formattedTime.append(minutes).append("m ")

        if (seconds > 0 && hours <= 0)
            formattedTime.append(seconds).append("s ")

        if (seconds <= 0 && minutes <= 0 && hours <= 0)
            formattedTime.append(ticks).append("t ")

        return formattedTime.trim().toString()
    }

    fun format(input: Long): String {
        val pattern = DecimalFormat("#.##")
        val suffixes = arrayOf(" ", " k", " M", " G", " T")

        var output = input

        for (suffix in suffixes) {
            if (output >= 1000)
                output /= 1000
            else
                return pattern.format(output) + suffix
        }

        return pattern.format(output) + suffixes[suffixes.size - 1]
    }
}